### PR TITLE
[AutoDiff] Move VJPEmitter out to separate file.

### PR DESCRIPTION
**Pull request by bgogul**
Refactoring for https://bugs.swift.org/browse/TF-993

### CODE REVIEW

**Comment by bgogul** [comment]
@swift-ci please test tensorflow

**Comment by bgogul** [comment]
@swift-ci please clean test tensorflow

**Review by dan-zheng — APPROVED** [review]

**Comment by bgogul** [comment]
@swift-ci please test tensorflow

File: include/swift/SILOptimizer/Utils/Differentiation/VJPEmitter.h [hunk 1]
```diff
@@ -0,0 +1,147 @@
+//===--- VJPEmitter.h - VJP Generation in differentiation -----*- C++ -*---===//
```
Comment (rxwei) [inline include/swift/SILOptimizer/Utils/Differentiation/VJPEmitter.h]:
```suggestion
//===--- VJPEmitter.h - VJP generation in differentiation -----*- C++ -*---===//
```

File: lib/SILOptimizer/Utils/Differentiation/VJPEmitter.cpp [hunk 2]
```diff
@@ -0,0 +1,666 @@
+//===--- VJPEmitter.cpp - VJP Generation in Differentiation ---*- C++ -*---===//
```
Comment (rxwei) [inline lib/SILOptimizer/Utils/Differentiation/VJPEmitter.cpp]:
```suggestion
//===--- VJPEmitter.cpp - VJP generation in differentiation ---*- C++ -*---===//
```

File: lib/SILOptimizer/Utils/Differentiation/JVPEmitter.cpp [hunk 3]
```diff
@@ -1,4 +1,4 @@
-//===--- JVPEmitter.h - JVP Generation in Differentiation -----*- C++ -*---===//
+//===--- JVPEmitter.cpp - JVP Generation in Differentiation ---*- C++ -*---===//
```
Comment (rxwei) [inline lib/SILOptimizer/Utils/Differentiation/JVPEmitter.cpp]:
```suggestion
//===--- JVPEmitter.cpp - JVP generation in differentiation ---*- C++ -*---===//
```

Please use consistent sentence casing in all headings.

Comment (bgogul) [reply lib/SILOptimizer/Utils/Differentiation/JVPEmitter.cpp]: Done.

**Comment by bgogul** [comment]
@swift-ci please test tensorflow

**Comment by bgogul** [comment]
@swift-ci please test tensorflow

**Comment by bgogul** [comment]
@swift-ci please test tensorflow macos

**Comment by bgogul** [comment]
@swift-ci please test tensorflow

**Pull request merged**